tests/integration/test_federation.py

```diff
@@ -0,0 +1,215 @@
+    return vote
+
+
+################################################################################
+# Tests here on down
+
+
+@pytest.mark.bdb
+@pytest.mark.genesis
+def test_elect_valid(federation_3):
+    [bx, (s0, s1, s2)] = federation_3
+    tx = input_single_create(bx[0])
+    process_tx(s0)
+    process_tx(s1)
+    process_tx(s2)
+    process_vote(s2, False)
+    for i in range(3):
+        assert bx[i].get_transaction(tx.id, True)[1] == 'undecided'
+    process_vote(s0, True)
+    for i in range(3):
+        assert bx[i].get_transaction(tx.id, True)[1] == 'undecided'
+    process_vote(s1, True)
+    for i in range(3):
+        assert bx[i].get_transaction(tx.id, True)[1] == 'valid'
+
+
+@pytest.mark.bdb
+@pytest.mark.genesis
+def test_elect_invalid(federation_3):
+    [bx, (s0, s1, s2)] = federation_3
+    tx = input_single_create(bx[0])
+    process_tx(s0)
+    process_tx(s1)
+    process_tx(s2)
+    process_vote(s1, True)
+    for i in range(3):
+        assert bx[i].get_transaction(tx.id, True)[1] == 'undecided'
+    process_vote(s2, False)
+    for i in range(3):
+        assert bx[i].get_transaction(tx.id, True)[1] == 'undecided'
+    process_vote(s0, False)
+    for i in range(3):
+        assert bx[i].get_transaction(tx.id, True)[1] is None
+
+
+@pytest.mark.bdb
+@pytest.mark.genesis
+def test_elect_disagree_prev_block(federation_3):
+    [bx, (s0, s1, s2)] = federation_3
+    tx = input_single_create(bx[0])
+    process_tx(s0)
+    process_tx(s1)
+    process_tx(s2)
+    process_vote(s0, True)
+    for i in range(3):
+        assert bx[i].get_transaction(tx.id, True)[1] == 'undecided'
+    s1.vote.last_voted_id = '5' * 64
+    process_vote(s1, True)
+    for i in range(3):
+        assert bx[i].get_transaction(tx.id, True)[1] == 'undecided'
+    s2.vote.last_voted_id = '6' * 64
+    process_vote(s2, True)
+    for i in range(3):
+        assert bx[i].get_transaction(tx.id, True)[1] is None
+
+
+@pytest.mark.skip()  # TODO: wait for #1309
+@pytest.mark.bdb
+@pytest.mark.genesis
+def test_elect_dupe_vote(federation_3):
+    from bigchaindb.exceptions import CriticalDuplicateVote
+    [bx, (s0, s1, s2)] = federation_3
+    tx = input_single_create(bx[0])
+    process_tx(s0)
+    process_tx(s1)
+    process_tx(s2)
+    vote = process_vote(s0, True)
+    # Drop the unique index and write the vote again
+    bx[0].connection.db.votes.drop_index('block_and_voter')
+    s0.queues['vote_write_vote'].append([vote])
+    s0.vote_write_vote()
+    for i in range(3):
+        with pytest.raises(CriticalDuplicateVote):
+            bx[i].get_transaction(tx.id, True)[1]
+
+
+@pytest.mark.bdb
+@pytest.mark.genesis
+def test_elect_sybill(federation_3):
+    [bx, (s0, s1, s2)] = federation_3
+    tx = input_single_create(bx[0])
+    process_tx(s0)
+    process_tx(s1)
+    process_tx(s2)
+    # What we need is some votes from unknown nodes!
+    for s in [s0, s1, s2]:
+        s.vote.bigchain.me_private = generate_key_pair()[0]
```

Comment:
This will only make the vote signatures invalid, not make the nodes ineligible which I think its what we want to test.